docs/Developer-Docs/VSCode-Integration.rst

```diff
@@ -0,0 +1,84 @@
+Visual Studio Code Integration
+------------------------------
+
+VSCode is a powerful IDE that can be used to do code and documentation
+development across the FireSim repository. It supports a client-server protocol
+over SSH that enables you to run a local GUI client that interacts with server
+running on your remote manager.
+
+
+General Setup
+=============
+
+#. Install VSCode. You can grab installers `here <https://code.visualstudio.com/download>`_.
+#. Open VSCode and install the ``Remote Developer Plugin``. See the `marketplace page <https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack>`_ for a complete description of its features.
+
+At this point, VSCode will slurp in your `.ssh/config`. Hosts you've listed
+there will be listed under the ``Remote Explorer`` in the left sidebar. You'll be able to connect to
+these hosts and create workspaces under firesim clones you've created there. 
+
+Workspace Locations
+===================
+
+Certain plugins assume the presence of certain files in particular locations,
+and often it is diserable to reduce the scope of files that VSCode will index.
+We recommend opening workspaces at the following locations:
+
+ * Scala and C++ development: ``sim/``
+ * RST docs: ``docs/``
+ * Manager (python): ``deploy/``
+
+You can always open a workspace at the root of FireSim -- just be cognizant that
+certain language-specific plugins (e.g., may not be configured correclty).
+
+
+Scala Development
+=========================
+
+.. warning:: Until Chipyard is bumped, you must add bloop to chipyard's plugins.sbt for this to work correctly. See :gh-file-ref:`sim/project/plugins.sbt` and copy the bloop installation into ``target-design/chipyard/project/plugins.sbt``.
+
+VSCode has rich support for Scala development, and the `Metals <https://scalameta.org/metals/docs/editors/vscode/>`_ plugin is really what makes the magic happen.
+
+How To Use (Remote Manager)
+###########################
+
+#. If you haven't already, clone FireSim and run build-setup.sh on your manager.
+#. Ensure your manager instance is listed as a host in your ``.ssh/config``. For example:
```

Comment:
You need this so that the remote explorer shows available hosts neatly. 

Personally, i have multiple managers set up and they are named such that most of their settings derive from a single wildcard pattern.